packages/cozy-harvest-lib/src/models/ConnectionFlow.spec.js

```diff
@@ -369,138 +368,92 @@ describe('ConnectionFlow', () => {
+      await flow.handleFormSubmit({ konnector: fixtures.clientKonnector })
+      expect(window.ReactNativeWebView.postMessage).toHaveBeenCalledWith(
+        JSON.stringify({
+          message: 'startLauncher',
+          value: {
+            connector: fixtures.clientKonnector
           }
-        }
-      }
-
-      it('should use custom konnector policy', async () => {
-        saveAccount.mockReset().mockImplementation((konnector, acc) => ({
-          ...acc,
-          _id: fixtures.updatedAccount._id
-        }))
-
-        biKonnectorPolicy.onAccountCreation
-          .mockReset()
-          .mockReturnValue(onAccountCreationResult)
-
-        const { flow, client } = setup()
-
-        jest
-          .spyOn(flow, 'ensureTriggerAndLaunch')
-          .mockResolvedValue(fixtures.launchedJob)
-
-        await setupSubmit(flow, {
-          konnector: bankingKonnector,
-          account: bankingAccount,
-          userCredentials: fixtures.bankingKonnectorAccountAttributes.auth
         })
+      )
 
-        expect(biKonnectorPolicy.onAccountCreation).toHaveBeenCalledTimes(1)
-        expect(flow.ensureTriggerAndLaunch).toHaveBeenCalledTimes(1)
-        expect(saveAccount).toHaveBeenCalledWith(
-          client,
-          flow.konnector,
-          onAccountCreationResult
-        )
-      })
-    })
-  })
-
-  describe('ensureTriggerAndLaunch', () => {
-    beforeAll(() => {
-      jest.spyOn(cronHelpers, 'fromFrequency').mockReturnValue('0 0 0 * * 0')
-      watchKonnectorJob.mockReturnValue({ on: () => ({}) })
-    })
-
-    afterEach(() => {
-      jest.clearAllMocks()
+      delete window.cozy
+      delete window.ReactNativeWebView
     })
 
-    it('should launch trigger without account', async () => {
-      const { flow, client } = setup()
-      await flow.ensureTriggerAndLaunch(client, {
-        trigger: fixtures.createdTrigger,
-        konnector: fixtures.konnector
-      })
-      expect(launchTrigger).toHaveBeenCalledTimes(1)
-      expect(launchTrigger).toHaveBeenCalledWith(
+    it('should not fail when no launcher is available', async () => {
+      const { client } = setup()
+      const flow = new ConnectionFlow(
         client,
-        fixtures.createdTrigger
+        fixtures.existingTrigger,
+        fixtures.clientKonnector
       )
-    })
 
-    it('should launch trigger with account', async () => {
-      const { flow, client } = setup()
-      ensureTrigger.mockResolvedValue(fixtures.existingTrigger)
-      await flow.ensureTriggerAndLaunch(client, {
-        account: fixtures.existingAccount,
-        trigger: fixtures.existingTrigger,
-        konnector: fixtures.konnector
-      })
-      expect(launchTrigger).toHaveBeenCalledTimes(1)
-      expect(launchTrigger).toHaveBeenCalledWith(client, {
-        ...fixtures.existingTrigger
+      const promise = flow.handleFormSubmit({
+        konnector: fixtures.clientKonnector
       })
+      expect(async () => await promise).not.toThrow()
```

Comment:
but this time, should we test that `launch` is called?

Reply:
If no launcher is available, the trigger is not launched

Reply:
I don't understand. If there is no launcher, then the policy will be the default one, right? And then, `needsTriggerLaunch` will be true, no?

Reply:
We are in case like this : in a web browser, in the home application, on a clisk.

- Since this is a web browser, there is no launcher available
- Since this is a clisk, the clisk policy is chosen

-> The trigger is not launched. Actually, the button to launch the konnector is not even enabled

I will even add a test to check that the trigger is not launched

Reply:
OK I got it, this is because of `konnector: fixtures.clientKonnector` 🤦